### PR TITLE
Distinct kill and restart actions

### LINKED ISSUE #20: Possible to kill process but not auto restart?
This would an ideal tool for our QA team for testing events that happen after the process has been destroyed. Things like handling push notifications that trigger a cold start for instance.

When I kill process, it auto restarts. Is there a way to block the restart part of it?

### PR DESCRIPTION
**Pull request by SimonMarquis**
- kill: simply kills the process
- restart: kills the process and restarts it

Reorder the notification actions to Restart > Kill > Cancel.
This order feels more natural for what this library tries to achieve.

Fixes #20

### CODE REVIEW

**Comment by SimonMarquis** [comment]
We could also update [preview/header.png](preview/header.png) and  [preview/preview.gif](preview/header.gif).

**Comment by SimonMarquis** [comment]
Waiting for #35 to be merged.

File: venom/src/main/java/com/github/venom/service/VenomService.kt [hunk 1]
```diff
@@ -49,11 +50,17 @@ internal class VenomService : Service() {
...
-            ACTION_KILL -> launchDeath()
+            ACTION_KILL -> kill()
+            ACTION_RESTART -> launchDeath()
         }
         return START_NOT_STICKY
     }
 
+    private fun kill() {
+        stopSelf()
+        Process.killProcess(Process.myPid())
```
Comment (YarikSOffice) [inline venom/src/main/java/com/github/venom/service/VenomService.kt]: Hey Simon! Thanks for the contribution

According to my testing, the new button still restarts the app. Perhaps, I'm missing something - let's try to figure it out.

Have you run the sample app to test this solution?
It looks like you are using the same function to kill the app as we do for the kill-restart scenario. Any difference?

Comment (SimonMarquis) [reply venom/src/main/java/com/github/venom/service/VenomService.kt]: My bad, I tested it once and assumed it worked. Unfortunately the OS tries to be "smart" and restart the process if it detects an Activity was not correctly stopped.

**Comment by SimonMarquis** [comment]
But... I think I found a workaround that could be used to really simulate a kill (without a restart).

The trick is to start a new `Activity` with the [FLAG_ACTIVITY_TASK_ON_HOME](https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_TASK_ON_HOME) and a "unique" [android:taskAffinity](https://developer.android.com/guide/topics/manifest/activity-element#aff).

This will have the effect of starting this new Activity "from" the launcher, and killing the process at this point will not re-start the whole application.

This can also be tested by starting the launcher and immediately killing the process, though it can bring the intent chooser:

```kotlin
val launcher = Intent(Intent.ACTION_MAIN).apply { 
    addCategory(Intent.CATEGORY_HOME)
    flags = Intent.FLAG_ACTIVITY_NEW_TASK
}
startActivity(launcher)
```


Here is what it looks like with an intermediate Activity:
As you can seen, the original activity stack is restored, and the process id has changed.

https://user-images.githubusercontent.com/1921278/173668610-69f9f629-b634-4b7a-a9ef-b9b40ee110bd.mp4

**Comment by YarikSOffice** [comment]
@SimonMarquis could you please share a code snippet to achieve this behavior? 

I'm trying to replicate this and add finally this feature if your solution is stable enough.

**Comment by SimonMarquis** [comment]
Hi @YarikSOffice, here is the updated version.
Please tell me if you can reproduce this behavior on your end.

Also, I've taken the liberty of renaming `DeathActivity` to `VenomActivity` since it can host both kill & restart actions without any code change.

**Comment by SimonMarquis** [comment]
Hi Yaroslav, had some time to review this?
Can you reproduce it on your devices?
Tell me if you need more info.

**Comment by YarikSOffice** [comment]
@SimonMarquis Hey! 
Let's try to stabilize this solution and maybe add new tests. If this doesn't work, I think there is another way to kill the process without triggering a restart.

**Comment by SimonMarquis** [comment]
Hi @YarikSOffice,
I've found the issue. It was something I did not test during my expermientations.
I initially thought we could use the same Activity for both kill & restart actions, but it turns out `android:taskAffinity` will prevent the restart action to properly restart the Activity.
Well therefore have to split both Activities, but it can be done in a minimalist way.

**Comment by YarikSOffice** [comment]
@SimonMarquis curious what is the reason for the activity not being restarted properly? in the case of task affinity

i didn't see any issues while making tests

**Comment by SimonMarquis** [comment]
I think forcing a different task activity would lead to 2 tasks being created (1 with your app activity,  1 with the Venom Activity as a root) when the kill action is requested.
And I guess the system only tries to restat the most recent tasks, which is now empty. 🤷

**Comment by SimonMarquis** [comment]
I'd love to add tests for this action as well, but could you take a look at #37 before, and merge it if you are ok with these changes?

**Comment by YarikSOffice** [comment]
I merged the PR with tests. but it seems like the build is failing now

**Comment by SimonMarquis** [comment]
Really strange to be honest. Can you re-launch the checks?

**Comment by SimonMarquis** [comment]
Also, since the new tests will be kind of copies of the previous tests, would you be ok if I migrate to [TestParameterInjector](https://github.com/google/TestParameterInjector)?

It would simplify the test matrix _a lot_ with a single test method.

**Comment by YarikSOffice** [comment]
I'm currently testing another way to get it working

https://github.com/YarikSOffice/venom/pull/41

**Comment by YarikSOffice** [comment]
[TestParameterInjector](https://github.com/google/TestParameterInjector) looks good to me

**Comment by YarikSOffice** [comment]
Thanks for your contribution Simon!

I'm closing this PR in favor of https://github.com/YarikSOffice/venom/pull/41 which is heavily based on yours. Really appreciate your efforts 👍 
Eventually, I decided to avoid the test matrix since it complicates debugging if a specific test is failed.

Sorry for the late resolution of this PR.

**Pull request closed without merging**